design-library/src/components/BccReact/BccReact.vue

```diff
@@ -0,0 +1,82 @@
+<script setup lang="ts">
+import { AddReactionFillIcon, AddReactionIcon, KeyboardArrowLeftIcon } from "@bcc-code/icons-vue";
+import { computed, ref } from "vue";
+import BccReactEmoji from "./BccReactEmoji.vue";
+import type { EmojiStat } from "./BccReactEmoji.vue";
+
+type Props = {
+  emojis: EmojiStat[];
+  top?: boolean;
+};
+
+const props = defineProps<Props>();
+const emit = defineEmits<{
+  (e: "toggle", id: string): void;
+}>();
+
+const activeEmojis = computed(() => {
+  const active = props.emojis.filter((emoji) => emoji.count && emoji.count > 0);
+  active.sort((a, b) => b.count! - a.count!);
+  return active;
+});
+
+const show = ref(false);
+function selectEmoji(emoji: EmojiStat) {
+  emit("toggle", emoji.id);
+
+  setTimeout(() => {
+    show.value = false;
+  }, 250);
+}
+</script>
+
+<template>
+  <div class="relative flex w-full items-center overflow-visible">
+    <TransitionGroup name="bcc-fade">
+      <button
+        key="toggle"
+        @click="show = !show"
+        v-if="show || emojis.some((e) => !e.selected)"
+        class="mr-1 flex shrink-0 cursor-pointer items-center justify-center rounded-full p-1 leading-tight transition"
+        :class="[top ? 'rounded-b-full' : 'rounded-t-full']"
+      >
+        <AddReactionFillIcon v-if="show" class="w-6" />
+        <AddReactionIcon v-else class="w-6" />
+      </button>
+
+      <div
+        class="hide-scrollbar flex flex-1 items-center gap-1 overflow-x-auto overflow-y-hidden rounded-full p-1"
+      >
+        <template v-if="activeEmojis.length > 0">
+          <TransitionGroup name="bcc-explode" appear>
+            <template v-for="emoji in activeEmojis" :key="emoji.id">
+              <keep-alive>
+                <BccReactEmoji @click="selectEmoji(emoji)" v-bind="emoji" />
+              </keep-alive>
+            </template>
+          </TransitionGroup>
+        </template>
+        <p v-else class="text-label flex items-center">
+          <KeyboardArrowLeftIcon class="mr-1 w-4" /> Be the first to react 😉
```

Comment:
This text should probably be configurable - and when it's empty the icon should be hidden as well.

Reply:
As a general note, I think it's fine to put it as a prop for now, for example the `BccInput` has an `optionalLabel` prop as well (that just defaults to the string "Optional"). But for the future I feel there should be a global config for the library, so that you don't need to customise it for every instance of the component (mostly relevant for localisation probably), but can just put it once in some config.

Reply:
Added placeholder prop